### PR TITLE
[testing] check for n_jobs design conformity

### PR DESCRIPTION
**Pull request by icfaust**
# Description
This will test if the design rule of using 'control_n_jobs' and 'run_with_n_jobs' is followed for all patched classes

Changes proposed in this pull request:
- Add test for n_jobs

Tasks
- [x] Implement tests
- [ ] Correct Failures

### CODE REVIEW

**Comment by icfaust** [comment]
CI issues are dependent on work in PR #1622

File: sklearnex/tests/test_patching.py [hunk 1]
```diff
@@ -116,7 +118,65 @@ def _load_all_models(patched):
...
+def test_daal4py_run_with_n_jobs_decorated(name):
+    patched = PATCHED_MODELS[name]
+    funcs = [i for i in dir(patched) if inspect.isfunction(getattr(patched, i))]
+    for i in funcs:
+        text = inspect.getsource(getattr(patched, i))
+
+        # find if it contains @run_with_n_jobs
+        njobs = "@run_with_n_jobs" in text
+        # find if it contains a call to a daal4py function
+        onedal_method = len(re.findall(r"daal4py\.\S+\(", text)) > 0
```
Comment (Alexsandruss) [inline sklearnex/tests/test_patching.py:161]: `@run_with_n_jobs`-decorated function may contain daal4py call or nested calls in other function like it's done in TSNE/KMeans. Therefore, runtime code analysis is not usable for this check.

File: sklearnex/tests/test_patching.py [hunk 2]
```diff
@@ -116,7 +118,65 @@ def _load_all_models(patched):
...
+        "@control_n_jobs" in lines
+    ), f"'{name}' does not support sklearnex n_job design rules"
+
+
+@pytest.mark.parametrize(
+    "name", [i for i in PATCHED_MODELS if "daal4py" in PATCHED_MODELS[i].__module__]
+)
+def test_daal4py_run_with_n_jobs_decorated(name):
+    patched = PATCHED_MODELS[name]
+    funcs = [i for i in dir(patched) if inspect.isfunction(getattr(patched, i))]
```
Comment (Alexsandruss) [inline sklearnex/tests/test_patching.py:154]: Runtime code analysis might be replaced with logs check.

File: sklearnex/tests/test_patching.py (hunk 1 above)
Comment (icfaust) [reply sklearnex/tests/test_patching.py]: daal4py test has been removed.

**Comment by icfaust** [comment]
I will likely either refactor this PR or close it depending on the progress of other PRs

**Pull request closed without merging**